posthog/hogql/resolver.py

```diff
@@ -675,7 +675,10 @@ def visit_field(self, node: ast.Field):
                 new_node: ast.Expr = ast.Alias(alias=node.type.name, expr=new_expr, hidden=True)
 
                 if node.type.isolate_scope:
-                    self.scopes.append(ast.SelectQueryType(tables={node.type.name: node.type.table_type}))
+                    table_type = node.type.table_type
+                    while isinstance(table_type, ast.VirtualTableType):
+                        table_type = table_type.table_type
+                    self.scopes.append(ast.SelectQueryType(tables={node.type.name: table_type}))
```

Comment:
I also wonder if this should be tested somehow in `test_resolver.py`, though it might be tricky and I'm happy to 🙈 here if that helps

Reply:
What's the reason behind not wanting a virtual table attached to the scope? Will this break existing behavior elsewhere at all?

Reply:
> I also wonder if this should be tested somehow in `test_resolver.py`, though it might be tricky and I'm happy to 🙈 here if that helps

@mariusandra It's implicitly covered by 56f88bf98b68f20ffbceab0b2acad65bd838ae77 (new test), and that style of test is generally how I'd think to solve it (register a new database table and then verify the field resolves through a virtual table to it.

Is there some other style of test you had in mind?

Reply:
Yeah... I think it's fine. It would probably be very similar code in both places, so I'm happy with it as it is.